docusaurus/docs/React/custom-code-examples/video-integration-100ms.mdx

```diff
@@ -0,0 +1,881 @@
+      } else if (connectionState == ConnectionState.Connecting) {
+        setConnectionState(ConnectionState.InCall);
+      }
+    } else {
+      setConnectionState(ConnectionState.NoCall);
+    }
+  };
+
+  client.on('channel.updated', handleChannelUpdate);
+
+  return () => {
+    client.off('channel.updated', handleChannelUpdate);
+  };
+}, [connectionState]);
+```
+
+The dependency makes sure the closure gets an updated version of the `connectionState` object every time it changes and by returning a function we make sure that the subscription to the `channel.updated` events will be terminated on the destruction of the component.
+
+In order to handle the implementation of our first function (`createCall`), we need to import the `100ms` SDK because we want to join a call in this case.
+
+So let’s install this from the terminal:
+
+```tsx
+yarn add @100mslive/react-sdk@latest
+```
+
+We need to inject the `HMSRoomProvider` into our application. Open up `App.tsx` and wrap the `<Chat>` element with the `<HMSRoomProvider>` element:
+
+```tsx
+<HMSRoomProvider>
+  <Chat>/* ... */</Chat>
+</HMSRoomProvider>
+```
+
+Now, we can add the import of the `useHMSActions` at the top of the `VideoContext.tsx`:
+
+```tsx
+import { useHMSActions } from '@100mslive/react-sdk';
+```
+
+And below the imports of the `channel` and the `client` add another line:
+
+```tsx
+const hmsActions = useHMSActions();
+```
+
+Great, with that we are prepared to implement the `createCall` function. We will do four things:
+
+1. Set the state to `ConnectionState.Connecting`
+2. Create a call from the `channel` object
+3. Join the call using the `hmsActions`
+4. Update the channel data (which will trigger an update of our UI due to the `channel.updated` event being called)
+
+Inside of the `store` object replace the empty `createCall` implementation with this:
+
+```tsx
+createCall: async () => {
+  // 1.
+  setConnectionState(ConnectionState.Connecting);
+
+  // 2.
+  const response = await channel.createCall({
+    id: `call-${channel.cid}`,
+    type: 'video',
+  });
+
+  // 3.
+  await hmsActions.join({
+    authToken: `${response.token}`,
+    userName: client.user?.name ?? 'Unkown',
+  });
+
+  // 4.
+  await channel.updatePartial({
+    set: {
+      data: {
+        callActive: true,
+        callId: response.call.id,
+      },
+    },
+  });
+},
+```
+
+The `joinCall` function is pretty similar to the `createCall` function only that we’re now taking the `callId` we save in the channel `data` object and using it to call the `getCallToken` function on the `client` object. Then we join the call as before. Here is the code to replace the current (empty) `joinCall` function with:
+
+```tsx
+joinCall: async () => {
+  setConnectionState(ConnectionState.Connecting);
+
+  const response = await client.getCallToken(channel.data?.data?.callId);
+
+  await hmsActions.join({
+    authToken: `${response.token}`,
+    userName: client.user?.name ?? 'Unkown',
```

Comment:
```suggestion
    userName: client.user?.name || 'Unkown',
```
This is a nitpick - is there a possibility that the username could be an empty string?

Reply:
It could probably, potentially be an empty string and in that case it would use the `'Unknown'` one which I feel like is a tad nicer, than showing an empty string here. :) 

So, will change it. Good catch!